api/features/views.py

```diff
@@ -187,6 +189,53 @@ def get_serializer_context(self):
 
         return context
 
+    def apply_state_to_queryset(
+        self, query_data: dict[str, typing.Any], queryset: QuerySet[Feature]
+    ) -> QuerySet[Feature]:
+        if not query_data.get("environment"):
+            raise serializers.ValidationError(
+                "Environment is required in order to filter by state search or by state enabled"
+            )
+        state_enabled = query_data["state_enabled"]
+        state_search = query_data["state_search"]
+        environment_id = query_data["environment"]
+
+        filter_search_q = Q()
+        if state_search is not None:
+            filter_search_q = filter_search_q | Q(
+                feature_state_value__string_value__icontains=state_search,
+            )
+
+            if state_search.lower() in {"true", "false"}:
```

Comment:
```suggestion
            if (boolean_search := state_search.lower()) in {"true", "false"}:
```

Let's get the walrus involved here because it's the best thing to ever happen in python!

Reply:
I think it's more legible without the parenthesis and walrus operator, but I don't want to crush your infectious enthusiasm so I'll do it your way just this one time.

Reply:
Running the code I've found that it doesn't actually match what we need, which is a value of `True` or `False`, instead it simply holds the value of `state_search.lower()` which is not what we need.

But just to keep you happy here's a walrus all for yourself:

<img width="346" alt="image" src="https://github.com/Flagsmith/flagsmith/assets/222715/9681e083-5625-4303-958d-68499de8e7cf">

Reply:
I think we could fix it by moving the parenthesis... but my walrus needs have been appeased (for now), so do as you see fit.